[Term Entry] Python Strings .isdigit() #3278 (#3297)
Codecademy/docs: content/python/concepts/strings/terms/isdigit/isdigit.md

---
Title: '.isdigit()'

> Description: 'Takes in a string and returns True if all characters in the string are digits and there is at least one character, False otherwise. Digits include decimal characters and digits that need special handling, such as the compatibility superscript digits. This covers digits which cannot be used to form numbers in base 10, like the Kharosthi numbers.'

Reviewer: This description is just a copy of the python documentation, that is considered plagiarism. All docs content should be original. Please revise the description and the text on line 16 to use your own, original phrasing.

https://docs.python.org/3/library/stdtypes.html?highlight=isdigit#str.isdigit

Author: Hey @caupolicandiaz, I revised the description and text on line 16. (Preview below)
![image](https://github.com/Codecademy/docs/assets/58596929/5dbbd6fb-7887-43ba-9e72-7057d66f343a)

---
Title: '.isdigit()'

> Description: 'This method under string class that checks if all the elements in the string are digits, decimals also including special cases like compatibility superscript digits.Returns True in the above cases and also requires atleast one character in the string. Returns False otherwise.'

Reviewer: ```suggestion
Description: 'Checks if all the elements in the string are digits and returns a boolean flag.'
```

> 

Reviewer: ```suggestion
```

Reviewer: the metadata section cannot contain any empty lines

Subjects:
  - 'Data Science'
  - 'Computer Science'
Tags:
  - 'Strings'
  - 'Methods'
  - 'Functions'

> 

Reviewer: ```suggestion
```

CatalogContent:
  - 'learn-python-3'
  - 'paths/analyze-data-with-python'
---

> The `.isdigit()` method under string class that checks if all the elements in the string are digits, decimals also including special cases like compatibility superscript digits.Returns `True` in the above cases and also requires atleast one character in the string. Returns `False` otherwise.

Reviewer: ```suggestion
The **`.isdigit()`** method under the string class checks if all the elements in the string are digits, applicable elements also include special cases like compatibility superscript digits. The method returns `True` in the above cases and also requires at least one character in the string. Returns `False` otherwise.
```

## Syntax

```pseudo
string.isdigit()
```

## Example

> 

Reviewer: There should be an introductory sentence before the example

```py
my_string1 = "123"
my_string2 = "123AA"

print(my_string1.isdigit())
# Output: True

print(my_string2.isdigit())

> # Output: False

Reviewer: The output should be in a ```shell block

```

## Codebyte Example

> Use `.isdigit()` to check if `my_string` contains only digits, `without ignoring` the dashes, exclamation point, and newline:

Reviewer: ```suggestion
The code below is runnable and uses `.isdigit()` to check `my_string`:
```

```codebyte/python
my_string = "!--1234--\n"
print(my_string.isdigit())

> # Output: False

Reviewer: ```suggestion
```

Reviewer: the output should not be included because this is a codebyte
